src/scvi/dataloaders/_data_splitting.py

```diff
@@ -205,10 +243,16 @@ def __init__(
             self.n_train, self.n_val = validate_data_split_with_external_indexing(
                 self.adata_manager.adata.n_obs,
                 self.external_indexing,
+                self.data_loader_kwargs.pop("batch_size", settings.batch_size),
```

Comment:
.get is preferable over pop as it keeps the dictionary unchanged.

Reply:
yeah, for some reason get didnt work here